guava/src/com/google/common/hash/Murmur3_32HashFunction.java

```diff
@@ -391,20 +391,20 @@ public HashCode hash() {
   }
 
   private static long codePointToFourUtf8Bytes(int codePoint) {
-    return (((0xFL << 4) | (codePoint >>> 18)) & 0xFF)
```

Comment:
This is a bit subtle. Because a code point can't have more than 21 bits, it's true that there can be at most 3 nonzero bits left after shifting right by 18, so the `0xFF` mask is unnecessary. I think a comment to that effect would be helpful.

Reply:
This subtlety applies to all the cases, not just first byte.

This cannot be `verify`-d because that would probably be too expensive.

I can add an `assert` if you accept that in the code base (like here https://github.com/google/guava/blob/769e1e9c5ea1c6bddf14cec66b71d6824a8d28a6/guava/src/com/google/common/hash/LittleEndianByteArray.java#L81)
or i can add a comment

Reply:
Although there are a few `assert` statements in this project, I think we would prefer it to be just a comment.

I'm not sure I understand
> This subtlety applies to all the cases, not just first byte.

It seems to me that it's only the first byte that is subtle. In the other cases, we are explicitly extracting out 6 bits to fit into a byte after the high `10`. That seems comparatively straightforward. On the other hand, a similar subtlety does apply to the first byte of each of the other methods, if that's what you meant.

Reply:
> It seems to me that it's only the first byte that is subtle.

i agree. Nvrm.